app/src/main/java/com/example/songme/mediaplayer/MediaPlayerManager.kt

```diff
@@ -0,0 +1,113 @@
+package com.example.songme.mediaplayer
+
+import android.media.AudioManager
+import android.media.MediaPlayer
+import com.example.songme.data.model.Track
+import com.example.songme.service.MusicService
+import com.example.songme.service.ServiceContract
+import com.example.songme.utils.Constants.DEFAULT_POSITION
+import com.example.songme.utils.Constants.POSITION_VALUE_ONE
+
+class MediaPlayerManager private constructor(private val musicService: MusicService) :
+    MediaPlayerContract {
+
+    private var mediaPlayer: MediaPlayer? = null
+    private var tracks: List<Track>? = null
+    private var currentPosition: Int = 0
+    private var onMediaPlayChange: ServiceContract.OnMediaPlayChange? = null
+
+    fun setOnMediaChangeListener(onMediaPlayChange: ServiceContract.OnMediaPlayChange?) {
+        this.onMediaPlayChange = onMediaPlayChange
+    }
+
+    override fun playTrack() {
+        val track = currentPosition.let { tracks?.get(it) }
+        mediaPlayer?.apply {
+            stop()
+            release()
+        }
+        mediaPlayer = MediaPlayer()
+        mediaPlayer?.setAudioStreamType(AudioManager.STREAM_MUSIC)
+        try {
+            mediaPlayer?.apply {
+                setDataSource(track?.streamUrl)
+                prepareAsync()
+                setOnPreparedListener(musicService)
+            }
+        } catch (e: Exception) {
+            TODO("NOT DONE")
+        }
+        onMediaPlayChange?.onTrackChange(track)
+    }
+
+    override fun pauseTrack() {
+        if (mediaPlayer!!.isPlaying) {
+            mediaPlayer?.pause()
+            onMediaPlayChange?.onMediaStateChange(false)
+        } else {
+            mediaPlayer?.start()
+            onMediaPlayChange?.onMediaStateChange(true)
+        }
+    }
+
+    override fun nextTrack() {
+        currentPosition++
+        if (currentPosition == tracks?.size) {
+            currentPosition = DEFAULT_POSITION
+        }
+        playTrack()
+    }
+
+    override fun prevTrack() {
+        currentPosition--
+        if (currentPosition < DEFAULT_POSITION) {
+            currentPosition = tracks?.size?.minus(POSITION_VALUE_ONE) ?: 0
+        }
+        playTrack()
+    }
+
+    override fun changeLoop() {
+        TODO("Not yet implemented")
+    }
+
+    override fun changeShuffled() {
+        TODO("Not yet implemented")
+    }
+
+    override fun getTracks(): List<Track?>? = tracks
```

Comment:
sao lại có track null được 
```suggestion
    override fun getTracks(): List<Track>? = tracks.filterNotNull()
```